tests/unit/dialog/dialog_core.js

```diff
@@ -7,6 +7,37 @@
 // TODO add teardown callback to remove dialogs
 module("dialog: core");
 
+test( "markup structure", function() {
+	expect( 7 );
+	var element = $("<div></div>").dialog({
+			buttons: [{
+				text: "Ok",
+				click: $.noop
+			}]
+		}),
+		widget = element.dialog( "widget" ),
+		titlebar = widget.find( ".ui-dialog-titlebar" ),
+		close = titlebar.find( "button" ),
+		buttonpane = widget.find( ".ui-dialog-buttonpane" );
+
+	ok( widget.is( ".ui-dialog.ui-widget.ui-widget-content.ui-corner-all.ui-dialog-buttons" ),
+		"Widget has proper classes" );
+
+	ok( titlebar.is( ".ui-dialog-titlebar.ui-corner-all.ui-widget-header"),
+		"Titlebar has proper classes");
+
+	ok( titlebar.find( ".ui-dialog-title" ).length, "Title is .ui-dialog-title" );
+
+	ok( close.hasClass( "ui-dialog-titlebar-close" ), "Close button is .ui-dialog-titlebar-close" );
+
+	ok( element.is( ".ui-dialog-content.ui-widget-content" ), "Element is .ui-widget-content" );
+
+	ok( buttonpane.is( ".ui-dialog-buttonpane.ui-widget-content" ),
+		"Buttonpane is .ui-widget-content" );
+	ok( buttonpane.find( ".ui-dialog-buttonset" ).length, "buttonset is .ui-dialog-buttonset" );
+
+});
```

Comment:
Despite it perhaps being a bit verbose for the sake of maintainability i would be in favor of many `.hasClass` tests so that the order does not matter but we get good failure messages. Alternatively we could stick with `.is` but use an overly verbose message that lists every class it should have. Disadvantage is you dont know which class ( or all ) is the culprit in the failure. 

The reason I don't like testing the class attribute is in many cases its perfectly possible ( and perfectly valid ) for someone to add additional classes directly to their markup. This would cause an class attribute test to fail ( I know thats not a problem in this specific test but it just seems like this should not cause a failure in general )

@jzaefferer @scottgonzalez thoughts?